### ISSUE TITLE
Conditional rendering

### ISSUE THREAD

**Issue by dunklesToast** [bpampuch/pdfmake#1957]
Is there any way to do conditional rendering? For example I want to render an image only, if a variable exists. What's the best way to do this without manually pushing it later into the docDefinition?

**Comment by JonesM87** [comment]
You would just need to conditionally add this to the doc definition when defining it, then render it, that's how I've been doing it for document variants.

**Comment by dunklesToast** [comment]
You mean something like this? I tried to avoid this but if this is the only way i'll use it. thx man
```
if(data.image){
   docDefinition.content.push({
      image: data.image
   })
}
```

**Comment by JonesM87** [comment]
This library will just render what you tell it to, your conditional logic etc will need to happen in your own code before passing the docDef to the library for rendering. In my case I had two documents which varied depending on a user type flag, when defining the docDef I just defined or did not define the elements as needed with if statements as I went along, I did this on definition rather than pushing to it after creation as my documents were quite intricate. This approach worked great for me.
